cpu/lm4f120/periph/timer.c

```diff
@@ -35,132 +37,411 @@
+    unsigned int timer_base;
+    unsigned int timer_side = TIMER_A;
+
+    if (dev >= TIMER_NUMOF){
+        return ;
+    }
+
+    switch(dev){
+#if TIMER_0_EN
+    case TIMER_0:
+        timer_base = WTIMER0_BASE;
+        break;
+#endif
+#if TIMER_1_EN
+    case TIMER_1:
+        timer_base = WTIMER1_BASE;
+        break;
+#endif
+    default:
+        return; // unreachable
     }
+
+    ROM_TimerEnable(timer_base, timer_side);
 }
 
 void timer_stop(tim_t dev)
 {
-    if (dev == TIMER_0) {
-        ROM_TimerDisable(WTIMER0_BASE, TIMER_A);
+    unsigned int timer_base;
+    unsigned int timer_side = TIMER_A;
+
+    if (dev >= TIMER_NUMOF){
+        return;
     }
+
+    switch(dev){
+#if TIMER_0_EN
+    case TIMER_0:
+        timer_base = WTIMER0_BASE;
+        break;
+#endif
+#if TIMER_1_EN
+    case TIMER_1:
+        timer_base = WTIMER1_BASE;
+        break;
+#endif
+    default:
+        return; // unreachable
+    }
+
+    ROM_TimerDisable(timer_base, timer_side);
 }
 
 void timer_irq_enable(tim_t dev)
 {
-    if (dev == TIMER_0) {
-        ROM_IntEnable(INT_WTIMER0A);
-        ROM_TimerIntEnable(WTIMER0_BASE, TIMER_TIMA_TIMEOUT);
+    if (dev >= TIMER_NUMOF){
+        return;
     }
+
+    unsigned int timer_intbase;
+
+    switch(dev){
+#if TIMER_0_EN
+    case TIMER_0:
+        timer_intbase = INT_WTIMER0A;
+        break;
+#endif
+#if TIMER_1_EN
+    case TIMER_1:
+        timer_intbase = INT_WTIMER1A;
+        break;
+#endif
+    default:
+        return; // unreachable
+    }
+
+    ROM_IntPrioritySet(timer_intbase, 32);
+    ROM_IntEnable(timer_intbase);
 }
 
 void timer_irq_disable(tim_t dev)
 {
-    if (dev == TIMER_0) {
-        ROM_IntDisable(INT_WTIMER0A);
+    if (dev >= TIMER_NUMOF){
+        return;
+    }
+    unsigned int timer_base;
+    unsigned int timer_intbit = TIMER_TIMA_TIMEOUT;
+    unsigned int timer_intbase;
+
+    switch(dev){
+#if TIMER_0_EN
+    case TIMER_0:
+        timer_base = WTIMER0_BASE;
+        timer_intbase = INT_WTIMER0A;
+        break;
+#endif
+#if TIMER_1_EN
+    case TIMER_1:
+        timer_base = WTIMER1_BASE;
+        timer_intbase = INT_WTIMER1A;
+        break;
+#endif
+    default:
+        return; // unreachable
     }
+
+    ROM_IntEnable(timer_intbase);
+    ROM_TimerIntDisable(timer_base, timer_intbit);
 }
 
 void timer_reset(tim_t dev)
 {
-    if (dev == TIMER_0) {
-        /* Performs a software reset of a peripheral */
-        ROM_SysCtlPeripheralReset(SYSCTL_PERIPH_WTIMER0);
+    if (dev >= TIMER_NUMOF){
+        return;
     }
+
+    unsigned int sysctl_timer;
+
+    switch(dev){
+#if TIMER_0_EN
+    case TIMER_0:
+        sysctl_timer = SYSCTL_PERIPH_WTIMER0;
+        break;
+#endif
+#if TIMER_1_EN
+    case TIMER_1:
+        sysctl_timer = SYSCTL_PERIPH_WTIMER1;
+        break;
+#endif
+    default:
+        return; // unreachable
+    }
+
+    /* Performs a software reset of a peripheral */
+    ROM_SysCtlPeripheralReset(sysctl_timer);
 }
 
 #if TIMER_0_EN
```

Comment:
Regarding the ISR, I was thinking we should also have something like common **irq_handler**. This routine is called by individual Timer ISRs and handles all redundant work.  Like how other cpu like **stm32f4** etc have implemented. I think that's a pretty neat structure.